### ISSUE TITLE
Transition rST to MyST?

### ISSUE THREAD

**Issue by consideRatio** [dask/dask-gateway#427]
I've never really become comfortable with rST, but in the past Markdown wasn't a good option because it lacked the ability to reference other files etc in a good way. This has changed with MyST though.

In the JupyterHub org, we have a few projects that have successfully transitioned MyST markdown with Sphinx instead of rST with Sphinx, and it has been a great success in my mind. For me, it has become easier to maintain docs in those projects and I believe it helps our Jupyter ecosystem contributors contribute to docs a bit easier as most of anyone working with Jupyter knows Markdown already.

Would you be interested in piloting the adoption of MyST (with Sphinx) instead of rST (with Sphinx) in the docs for this project?

### Related
- [JupyterHub Helm chart use of MyST](https://github.com/jupyterhub/zero-to-jupyterhub-k8s/tree/main/doc)
- [MyST Markdown reference](https://myst-parser.readthedocs.io/en/latest/)

**Comment by martindurant** [comment]
I think most/all of the other dask projects use rst, so maybe you want to ask more generally at dask/community. I personally am not a great fan of rst, so if this does the job, it may just be what I've been wanting.

**Comment by TomAugspurger** [comment]
Agreed that it's worth mentioning in dask/community, but I don't think that should block a subproject from adopting myst. Things like the dask-sphinx-them will still work just fine, regardless of whether the source is in rst or markdown.

My experience with MyST has been very positive.

**Comment by consideRatio** [comment]
I think it can make a lot of sense to acquire some input on a organizational level as well, so I opened https://github.com/dask/community/issues/178. There is value to be consistent across the org, and piloting the use of MyST in this project would break such consistency.
